client/src/partials/patients/edit/PatientGroupService.js

```diff
@@ -0,0 +1,77 @@
+angular.module('bhima.services')
+.service('PatientGroupModal', PatientGroupModal);
+
+PatientGroupModal.$inject = ['$uibModal'];
+
+/**
+ * This service is responsible for configuring and exposing modal that can be
+ * used to configure a patients groups subscription.
+ *
+ * @todo remove callbacks in favour of propegating values through promise fufillment
+ */
+function PatientGroupModal(Modal) {
+  var service = this;
+
+  var modalConfig = {
+    animation : true,
+    size : 'md',
+    keyboard : false,
+    backdrop : 'static'
+  };
+
+  service.updateDebtor = updateDebtor;
+  service.updateGroupConfig = updateGroupConfig;
+
+  /**
+   * This function if responsible for configuring and openening a modal to update
+   * debtor groups given a patient and a complete callback
+   *
+   * @params Object     patientEntity
```

Comment:
I just recently found this out: the correct syntax of indicating parameters is `@param`, not `@params`.  Not blocking, but just for future reference.